src/test/java/dog/pawbook/logic/commands/AddDogCommandTest.java

```diff
@@ -0,0 +1,202 @@
+package dog.pawbook.logic.commands;
+
+import static dog.pawbook.testutil.Assert.assertThrows;
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.commons.core.GuiSettings;
+import dog.pawbook.logic.commands.exceptions.CommandException;
+import dog.pawbook.model.AddressBook;
+import dog.pawbook.model.Model;
+import dog.pawbook.model.ReadOnlyAddressBook;
+import dog.pawbook.model.ReadOnlyUserPrefs;
+import dog.pawbook.model.managedentity.Entity;
+import dog.pawbook.model.managedentity.dog.Dog;
+import dog.pawbook.testutil.DogBuilder;
+import javafx.collections.ObservableList;
+import javafx.util.Pair;
+
+public class AddDogCommandTest {
+
+    @Test
+    public void constructor_nullDog_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddDogCommand(null));
+    }
+
+    @Test
+    public void execute_dogAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingEntityAdded modelStub = new ModelStubAcceptingEntityAdded();
+        Dog validDog = new DogBuilder().build();
+
+        CommandResult commandResult = new AddDogCommand(validDog).execute(modelStub);
+
+        assertEquals(AddDogCommand.MESSAGE_SUCCESS + validDog, commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validDog), modelStub.entitiesAdded);
+    }
+
+    @Test
+    public void execute_duplicateDog_throwsCommandException() {
+        Dog validDog = new DogBuilder().build();
+        AddDogCommand addDogCommand = new AddDogCommand(validDog);
+        ModelStub modelStub = new ModelStubWithDog(validDog);
+
+        assertThrows(CommandException.class,
+                AddDogCommand.MESSAGE_DUPLICATE_DOG, () -> addDogCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        Dog apple = new DogBuilder().withName("Apple").build();
+        Dog bubbles = new DogBuilder().withName("Bubbles").build();
+        AddDogCommand addAppleCommand = new AddDogCommand(apple);
+        AddDogCommand addBubblesCommand = new AddDogCommand(bubbles);
+
+        // same object -> returns true
+        assertTrue(addAppleCommand.equals(addAppleCommand));
+
+        // same values -> returns true
+        AddDogCommand addAppleCommandCopy = new AddDogCommand(apple);
+        assertTrue(addAppleCommand.equals(addAppleCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addAppleCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addAppleCommand.equals(null));
+
+        // different dog -> returns false
+        assertFalse(addAppleCommand.equals(addBubblesCommand));
+    }
+
+    /**
```

Comment:
package private?

Reply:
This is referring to the `ModelStub` btw

Reply:
sorry im not quite sure what u mean? i just referenced the AddOwnerCommandTest for this

Reply:
As in because previously there was only one CommandTest so given ModelStub's nature it made sense to just have it as a private class of the Command, but going forward we probably want to abstract out a bit of these test code as well. The ModelStub should become a package private shared by all the CommandTests, it will be good to combine the tests under AddCommandTest.